index.bs

```diff
@@ -569,13 +569,22 @@ Within protected samples, the following constraints apply:
 - [=Temporal Delimiter OBUs=], [=Sequence Header OBUs=], and [=Frame Header OBUs=] (including within a [=Frame OBU=]), [=Redundant Frame Header OBUs=] and [=Padding OBUs=] SHALL be unprotected.
 - [=Metadata OBUs=] MAY be protected.
 - [=Tile Group OBUs=] and [=Frame OBUs=] SHALL be protected. Within [=Tile Group OBUs=] or [=Frame OBUs=], the following applies:
-    - [=BytesOfProtectedData=] SHALL be a multiple of 16 bytes.
-    - [=BytesOfProtectedData=] SHALL end on the last byte of the <code>decode_tile</code> structure (including any trailing bits).
-    - [=BytesOfProtectedData=] SHALL span all complete 16-byte blocks of the <code>decode_tile</code> structure (including any trailing bits).
+	- for the protected scheme <code>[=cbcs=]</code>:
+		- [=BytesOfProtectedData=] SHALL start on the first byte and end on the last byte of the <code>decode_tile</code> structure (including any trailing bits).
 
-    NOTE: As a result of the above, partial blocks are not used and it is possible that [=BytesOfProtectedData=] does not start at the first byte of the <code>decode_tile</code> structure, but some number of bytes following that.
-    - A subsample SHALL be created for each tile that has a <code>decode_tile</code> structure whose size (including any trailing bits) is larger or equal to 16 bytes. If it is less than 16 bytes, per the rules above, the <code>decode_tile</code> structure is not encrypted and the corresponding bytes SHOULD be included in the [=BytesOfClearData=] field of a surrounding subsample, if any.
-	- All other parts of [=Tile Group OBUs=] and [=Frame OBUs=] SHALL be unprotected.
+	    NOTE: As a result of the above, [=BytesOfProtectedData=] is not necessarily a multiple of 16 bytes and partial blocks are included in the [=BytesOfProtectedData=] range, but per [[!CENC]] the data is not be encrypted.
+
+	    - A subsample SHALL be created for each tile, even if its size is less than 16 bytes.
```

Comment:
What is the reason for this requirement? This is against CENC spec recommendation:

Subsample encryption entries SHOULD be as compactly represented as possible. For
example, instead of two entries with {15 clear, 0 protected}, {17 clear, 500 protected}, use one entry
of {32 clear, 500 protected}. If pattern-based encryption is used, then the pattern applies to the
protected byte range, BytesOfProtectedData; otherwise, all protected bytes are encrypted.

Reply:
As discussed in today's Codec WG call, there are 2 ways one could merge the small tile (size R < 16):
a) if the small tile is not the last one, (it has a clear header: the tile size), it could only be merged with the clear range of the next subsample.
b) if the small tile is the last one, it has no header (tile size is not coded), so ByteOfClearData is 0. It could be merged with the previous subsample. Additionally, if the previous subsample is of size S = 10*16*N+R', with R'<16, merging both protected ranges would mean that if R+R'>=16, the R' bytes of the first tile would now be encrypted, followed by the next 16-R' bytes also encrypted, only the last 16-(R+R') would remain in the clear.

The current feeling of the group is that merging subsamples makes it harder for implementation and makes independent processing of tiles not possible. The advantages in terms of reduced overhead and increased encryption seem small.

Reply:
I don't like it either.

I was thinking about making it explicit that the tiles with less than 16 bytes are not encrypted and allowing it to be merged with the next subsample. Something like this:

	- for the protected scheme <code>[=cbcs=]</code>:
		- A subsample SHALL be created for each tile that has a <code>decode_tile</code> structure whose size (including any trailing bits) is larger than or equal to 16 bytes, with [=BytesOfProtectedData=] starts on the first byte and ends on the last byte of the <code>decode_tile</code> structure (including any trailing bits).
		- For tiles that has a <code>decode_tile</code> structure whose size (including any trailing bits) is smaller than 16 bytes, the <code>decode_tile</code> structure is not encrypted and the corresponding bytes SHOULD be included in the [=BytesOfClearData=] field of a surrounding subsample, if any.
	    NOTE: As a result of the above, [=BytesOfProtectedData=] is not necessarily a multiple of 16 bytes and partial blocks are included in the [=BytesOfProtectedData=] range, but per [[!CENC]] the data is not be encrypted.

This makes it consistent to what we are doing on `cenc`. What do you think?

Reply:
The Codec WG discussed again your proposal to allow merging but there was no support for that proposal. The Codec WG decided to adopt this PR as is. I will merge this PR. If you strongly feel that your change should be discussed again please open a new issue.